### ISSUE TITLE
Max line length configuration is ignored

### ISSUE THREAD

**Issue by Nate-Wessel** [sqlfluff/sqlfluff#4825]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

My `max_line_length` configuration seems to be ignored. I based that part of my config file off of the example in the docs at https://docs.sqlfluff.com/en/stable/configuration.html#new-project-configuration

I have the max length set to 100, but it still reports errors where lines are longer than 80 (the default), but less than 100. 

This wasn't a problem in version 1.4.5 

### Expected Behaviour

It should read the configured `max_line_length`

### Observed Behaviour

Given error like `LT05: Line is too long (90 > 80).`

### How to reproduce

Using the config file below, lint 
```sql
SELECT
    my_field || 'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmn'::text
FROM my_table;
```

### Dialect

postgres

### Version

2.0.7

### Configuration

```
[sqlfluff]
dialect = postgres
max_line_length = 100
exclude_rules = layout.end-of-file, capitalisation.functions, aliasing.forbid, structure.using, ambiguous.join, convention.left_join
warnings = layout.spacing, layout.long_lines, layout.start_of_file

[sqlfluff:indentation]
tab_space_size = 4
indent_unit = space

aliasing = explicit

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Are you still getting this? I can't repeat this.

**Comment by tunetheweb** [comment]
It definitely works with a clean install and a fresh config:

```
(.venv) 8:48:34 ~/sources/sqlfluff main $ git checkout 2.0.7
Note: switching to '2.0.7'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at d19de0ec Bump to version 2.0.7 (#4799)
(.venv) 8:48:38 ~/sources/sqlfluff tags/2.0.7 $ sqlfluff lint test.sql    
== [test.sql] FAIL                                                                                                                                                                        
L:   1 | P:   1 | LT09 | Select targets should be on a new line unless there is
                       | only one select target.
                       | [layout.select_targets]
All Finished 📜 🎉!
(.venv) 8:48:44 ~/sources/sqlfluff tags/2.0.7 $ more test.sql 
SELECT
    my_field || 'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmn'::text
FROM my_table;
(.venv) 8:48:53 ~/sources/sqlfluff tags/2.0.7 $ more .sqlfluff 
[sqlfluff]
dialect = postgres
max_line_length = 100
exclude_rules = layout.end-of-file, capitalisation.functions, aliasing.forbid, structure.using, ambiguous.join, convention.left_join
warnings = layout.spacing, layout.long_lines, layout.start_of_file

[sqlfluff:indentation]
tab_space_size = 4
indent_unit = space

aliasing = explicit

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
```

Can you run lint with `-vv` option and see what `max_line_length` is set to there?

Is the other config in your .sqlfluff file working?

Do you have any other nested .sqlfluff config files or only the one?

What command are you calling?

**Comment by Nate-Wessel** [comment]
@tunetheweb So sorry for the mistake; I looked into this a bit further and you're 100% correct. I was running this as a reusable Github action and had made a branch for the upgrade to SQLFluff 2... but it turns out I was still pulling my config file from the main branch which was in the v1 format. No warning abut a missing config file, as it was there but just in the old format. 

Apologies again, and thanks for your help!
